src/index.ts

```diff
@@ -13,5 +13,211 @@
  *  See the License for the specific language governing permissions and
  *  limitations under the License.
  ******************************************************************************* */
+import { ResponseSign, type ResponseAddress, type ResponsePubkey } from "./types";
+import { CLA, P1_VALUES, P2_VALUES, PKLEN } from "./consts";
+import type Transport from "@ledgerhq/hw-transport";
+import BaseApp, {
+  INSGeneric,
+  LedgerError,
+  PAYLOAD_TYPE,
+  ResponsePayload,
+  processErrorResponse,
+  processResponse,
+} from "@zondax/ledger-js";
 
-export * from './cosmosApp'
+const crypto = require("crypto");
+const bech32 = require("bech32");
+const Ripemd160 = require("ripemd160");
+
+export default class CosmosApp extends BaseApp {
+  static _INS = {
+    GET_VERSION: 0x00 as number,
+    SIGN_SECP256K1: 0x02 as number,
+    GET_ADDR_SECP256K1: 0x04 as number,
+  };
+
+  static _params = {
+    cla: CLA,
+    ins: { ...CosmosApp._INS } as INSGeneric,
+    p1Values: { ONLY_RETRIEVE: 0x00 as 0, SHOW_ADDRESS_IN_DEVICE: 0x01 as 1 },
+    chunkSize: 250,
+    requiredPathLengths: [5],
+  };
+
+  /**
+   * Constructs a new CosmosApp instance.
+   * @param transport - The transport instance.
+   * @throws {Error} - If the transport is not defined.
+   */
+  constructor(transport: Transport) {
+    super(transport, CosmosApp._params);
+
+    if (!this.transport) {
+      throw new Error("Transport has not been defined");
+    }
+  }
+
+  private serializeHRP(hrp: string) {
+    if (hrp == null || hrp.length < 3 || hrp.length > 83) {
+      throw new Error("Invalid HRP");
+    }
+    const buf = Buffer.alloc(1 + hrp.length);
```

Comment:
You can now move to ByteStream.
This simplifies a lot writing to the buffer and you will not need to preallocate.